deptofdefense/anet
# Adding 404 pages

Address #218.

![ANET screenshot](https://cloud.githubusercontent.com/assets/829827/23367494/7221982a-fcd8-11e6-9d8a-b330ce822f9d.png)

## To Test
Observe the 404 page on the following URLs:

1. http://localhost:3000/asdfasd
1. http://localhost:3000/people/445555
1. http://localhost:3000/people/445555asdf
1. http://localhost:3000/people/445555/edit
1. http://localhost:3000/people/445555asdf/edit
1. http://localhost:3000/locations/37333
1. http://localhost:3000/locations/37333asdf
1. http://localhost:3000/poams/37333
1. http://localhost:3000/poams/37333asdf
1. http://localhost:3000/poams/37333/edit
1. http://localhost:3000/poams/37333adsf/edit
1. http://localhost:3000/positions/3333
1. http://localhost:3000/positions/3333asdf
1. http://localhost:3000/positions/3333/edit
1. http://localhost:3000/positions/3333asdf/edit
1. http://localhost:3000/reports/301
1. http://localhost:3000/reports/30asdf
1. http://localhost:3000/reports/301/edit
1. http://localhost:3000/reports/30asfd/edit
1. http://localhost:3000/organizations/0
1. http://localhost:3000/organizations/0asdf
1. http://localhost:3000/organizations/0/edit
1. http://localhost:3000/organizations/0asdf/edit

Observe that the 404 page does not show up on the following URLs:

1. http://localhost:3000/
1. http://localhost:3000/people/44
1. http://localhost:3000/people/44/edit
1. http://localhost:3000/locations/37
1. http://localhost:3000/poams/61
1. http://localhost:3000/poams/61/edit
1. http://localhost:3000/positions/42/edit
1. http://localhost:3000/positions/42
1. http://localhost:3000/reports/30
1. http://localhost:3000/reports/30/edit
1. http://localhost:3000/organizations/51
1. http://localhost:3000/organizations/51/edit

## Known Issues
**Resolved** ~~I was not able to add 404 pages for organizations, because the backend returns a `500` error when you make a graphql query for an unknown id.~~

I am happy to accept feedback from @efo-usds or @readywater on the wording for the "not found" text.

**Resolved** ~~Some of this is messy / duplicative. Instead of fixing this in the client, I would rather make the backend behave consistently. I would recommend returning `404` or `null` consistently instead of returning `null` sometimes and `500` other times. @hunterp what do you think about this?~~

**Resolved** ~~When you navigate to page that should show a model (like `/reports/:id/edit`), you'll see a flash of the original page before it's removed and replaced with the 404 page. @erincd is this worth fixing?~~ **Answer:** Not right now.
